identity/src/main/java/com/android/identity/Util.java

```diff
@@ -298,15 +298,39 @@ static boolean cborDecodeBoolean(@NonNull byte[] data) {
         return simple.getSimpleValueType() == SimpleValueType.TRUE;
     }
 
+    /**
+     * Accepts a {@code DataItem}, attempts to cast it to a {@code Number}, then returns the value
+     * Throws {@code IllegalArgumentException} if the {@code DataItem} is not a {@code Number}. This
+     * method also checks bounds, and if the given data item is too large to fit in a long, it
+     * throws {@code ArithmeticException}.
+     */
+    static long asLong(DataItem item) {
+        BigInteger bigNum = castTo(Number.class, item).getValue();
+        if (bigNum.bitLength() > 64) {
```

Comment:
I have no idea whether this is the right condition.

WDYT of changing the logic to something that is obviously correct (but produces a temporary object), e.g.:

~~~
BigInteger bigInteger = castTo(Number.class, item).getValue();
long result = bigInteger.longValue();
if (!(BigInteger.valueOf(result).equals(bigInteger)) {
  throw new ArithmeticException("Overflow: doesn't fit into a long: " + bigInteger);
}
~~~

?

We shoul dalso add unit tests covering the boundary cases - I'll leave a comment in UtilTest.java

Reply:
I have no idea why, but it turns out this condition was indeed wrong!

I improved the unit test to better check boundary conditions.